[BUG]: Error Compiling for board Teensy 4.1

**Describe the bug**
Not able to compile the "BasicReadAngle" example.

**Error Message**
Arduino: 1.8.19 (Windows 10), TD: 1.57, Board: "Teensy 4.1, Serial, 600 MHz, Faster, US English"

In file included from D:\0-Files\Documents\Arduino\libraries\AS5047P\src\types\AS5047P_Types.cpp:12:0:

D:\0-Files\Documents\Arduino\libraries\AS5047P\src\types\AS5047P_Types.h:140:18: error: 'string' in namespace 'std' does not name a type

             std::string toStdString();

                  ^

D:\0-Files\Documents\Arduino\libraries\AS5047P\src\types\AS5047P_Types.cpp:35:38: error: no 'std::__cxx11::string AS5047P_Types::ERROR_t::toStdString()' member function declared in class 'AS5047P_Types::ERROR_t'

     std::string ERROR_t::toStdString() {

                                      ^

Error compiling for board Teensy 4.1.



This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.

--- a/Doxyfile
+++ b/Doxyfile
@@ -38,7 +38,7 @@ PROJECT_NAME           = "AS5047P - Arduino Library"
 # could be handy for archiving the generated documentation or if some version
 # control system is used.
 
-PROJECT_NUMBER         = 2.1.5
+PROJECT_NUMBER         = 2.2.0
 
 # Using the PROJECT_BRIEF tag one can provide an optional one line description
 # for a project that appears at the top of each page and should give viewer a

--- a/examples/BasicReadAngle/BasicReadAngle.ino
+++ b/examples/BasicReadAngle/BasicReadAngle.ino
@@ -3,8 +3,8 @@
  * @author Jonas Merkle [JJM] (jonas@jjm.one)
  * @brief This is a basic example program to read the angle position from a AS5047 rotary encoder.
  *        The angle postion gets updated and printed to the serial consol once a second. 
- * @version 2.1.5
- * @date 2021-04-10
+ * @version 2.2.0
+ * @date 2022-11-20
  * 
  * @copyright Copyright (c) 2021 Jonas Merkle. This project is released under the GPL-3.0 License License.
  * 

--- a/examples/BasicReadAngleAndDebugInfo/BasicReadAngleAndDebugInfo.ino
+++ b/examples/BasicReadAngleAndDebugInfo/BasicReadAngleAndDebugInfo.ino
@@ -3,8 +3,8 @@
  * @author Jonas Merkle [JJM] (jonas@jjm.one)
  * @brief This is a basic example program to read the angle position and debug information from a AS5047 rotary encoder.
  *        The angle postion and debug information gets updated and printed to the serial consol once a second. 
- * @version 2.1.5
- * @date 2021-04-10
+ * @version 2.2.0
+ * @date 2022-11-20
  * 
  * @copyright Copyright (c) 2021 Jonas Merkle. This project is released under the GPL-3.0 License License.
  * 
@@ -50,16 +50,8 @@ void loop() {
   digitalWrite(LED_PIN, HIGH);                    // activate the led.
   Serial.print("Angle: ");                        // print some text to the serial consol.
   Serial.println(as5047p.readAngleDegree());      // read the angle value from the AS5047P sensor an print it to the serial consol.
-  #ifdef ARDUINO_ARCH_SAMD
-  std::string s = as5047p.readStatusAsStdString();      // get the string containing the debug information.
-  for(uint16_t i = 0; i < s.size(); i++) {              // printing the string character by character
-    Serial.print(s[i]); 
-  }
-  Serial.println("");
-  #else
   Serial.println(as5047p.readStatusAsArduinoString());  // get the string containing the debug information and print it.
   Serial.println("");
-  #endif
   delay(500);                                     // wait for 500 milli seconds.
 
   // wait

--- a/examples/BasicReadAngleWithErrorInfo/BasicReadAngleWithErrorInfo.ino
+++ b/examples/BasicReadAngleWithErrorInfo/BasicReadAngleWithErrorInfo.ino
@@ -4,8 +4,8 @@
  * @brief This is a basic example program to read the angle position from a AS5047 rotary encoder.
  *        The angle postion gets updated and printed to the serial consol once a second.
  *        In addition to the angle value the error information gets displayed.
- * @version 2.1.5
- * @date 2021-04-10
+ * @version 2.2.0
+ * @date 2022-11-20
  * 
  * @copyright Copyright (c) 2021 Jonas Merkle. This project is released under the GPL-3.0 License License.
  * 

--- a/examples/PrintAllSettings/PrintAllSettings.ino
+++ b/examples/PrintAllSettings/PrintAllSettings.ino
@@ -2,8 +2,8 @@
  * @file PrintAllSettings.ino
  * @author Jonas Merkle [JJM] (jonas@jjm.one)
  * @brief This is a example prints out the settings of the AS5047P sensor.
- * @version 2.1.5
- * @date 2021-04-10
+ * @version 2.2.0
+ * @date 2022-11-20
  * 
  * @copyright Copyright (c) 2021 Jonas Merkle. This project is released under the GPL-3.0 License License.
  * 

--- a/library.json
+++ b/library.json
@@ -1,6 +1,6 @@
 {
     "name": "AS5047P",
-    "version": "2.1.5",
+    "version": "2.2.0",
     "description": "An Arduino library for the AS5047P high-resolution rotary position sensor. Supporting also the following sensor types: AS5047D, AS5147, AS5147P, AS5247",
     "keywords": "arduino-library, as5047p, spi, sensor, rotary-encoder, high-resolution, rotary-position-sensor, as5x47, as5047d, as5147, as5147p, as5247, encoder, rotaty, angle, uvw, abi",
     "repository":

--- a/library.properties
+++ b/library.properties
@@ -1,5 +1,5 @@
 name=AS5047P
-version=2.1.5
+version=2.2.0
 author=Jonas Merkle [JJM] <jonas@jjm.one>
 maintainer=Jonas Merkle [JJM] <jonas@jjm.one>
 sentence=An Arduino library for the AS5047P high-resolution rotary position sensor.

--- a/src/AS5047P.cpp
+++ b/src/AS5047P.cpp
@@ -2,8 +2,8 @@
  * @file AS5047P.h
  * @author Jonas Merkle [JJM] (jonas@jjm.one)
  * @brief This is the main sourcefile of the AS5047P Library.
- * @version 2.1.5
- * @date 2021-04-10
+ * @version 2.2.0
+ * @date 2022-11-20
  * 
  * @copyright Copyright (c) 2021 Jonas Merkle. This project is released under the GPL-3.0 License License.
  * 
@@ -128,7 +128,7 @@ bool AS5047P::verifyWittenRegF(uint16_t regAddress, uint16_t expectedData) {
     return recData.data.raw == expectedData;
 }
 
-#if defined(ARDUINO_ARCH_SAMD) || defined(CORE_TEENSY)
+#if defined(AS5047P_STD_STRING_SUPPORT)
 
 std::string AS5047P::readStatusAsStdString() {
 

--- a/src/AS5047P.h
+++ b/src/AS5047P.h
@@ -2,8 +2,8 @@
  * @file AS5047P.h
  * @author Jonas Merkle [JJM] (jonas@jjm.one)
  * @brief This is the main headerfile of the AS5047P Library.
- * @version 2.1.5
- * @date 2021-04-10
+ * @version 2.2.0
+ * @date 2022-11-20
  * 
  * @copyright Copyright (c) 2021 Jonas Merkle. This project is released under the GPL-3.0 License License.
  * 
@@ -14,12 +14,14 @@
 
 #include <inttypes.h>
 
+#include "./util/AS5047P_Settings.h"
+
 //#ifdef Arduino_h
 #include <Arduino.h>
 #include "spi/AS5047P_SPI_Arduino.h"
 //#endif
 
-#if defined(ARDUINO_ARCH_SAMD) || defined(CORE_TEENSY)
+#if defined(AS5047P_STD_STRING_SUPPORT)
 #include <string>
 #endif
 
@@ -88,7 +90,7 @@ class AS5047P {
         bool verifyWittenRegF(uint16_t regAddress, uint16_t expectedData);
 
 
-        #if defined(ARDUINO_ARCH_SAMD) || defined(CORE_TEENSY)
+        #if defined(AS5047P_STD_STRING_SUPPORT)
         /**
          * Reads all status information from the AS5047P sensor and returns them as a string.
          * @return A std::string with all status information.

--- a/src/spi/AS5047P_SPI_Arduino.cpp
+++ b/src/spi/AS5047P_SPI_Arduino.cpp
@@ -4,8 +4,8 @@
  * @file AS5047P_SPI_Arduino.cpp
  * @author Jonas Merkle [JJM] (jonas@jjm.one)
  * @brief This sourefile contains the implementation of the Arduino SPI bus handler for the AS5047P Library.
- * @version 2.1.5
- * @date 2021-04-10
+ * @version 2.2.0
+ * @date 2022-11-20
  * 
  * @copyright Copyright (c) 2021 Jonas Merkle. This project is released under the GPL-3.0 License License.
  * 

--- a/src/spi/AS5047P_SPI_Arduino.h
+++ b/src/spi/AS5047P_SPI_Arduino.h
@@ -2,8 +2,8 @@
  * @file AS5047P_SPI_Arduino.h
  * @author Jonas Merkle [JJM] (jonas@jjm.one)
  * @brief This headerfile contains the Arduino SPI bus handler for the AS5047P Library.
- * @version 2.1.5
- * @date 2021-04-10
+ * @version 2.2.0
+ * @date 2022-11-20
  * 
  * @copyright Copyright (c) 2021 Jonas Merkle. This project is released under the GPL-3.0 License License.
  * 

--- a/src/types/AS5047P_Types.cpp
+++ b/src/types/AS5047P_Types.cpp
@@ -2,8 +2,8 @@
  * @file AS5047P_Types.cpp
  * @author Jonas Merkle [JJM] (jonas@jjm.one)
  * @brief This sourcefile contains the implementation of the type definitions for the AS5047P Library.
- * @version 2.1.5
- * @date 2021-04-10
+ * @version 2.2.0
+ * @date 2022-11-20
  * 
  * @copyright Copyright (c) 2021 Jonas Merkle. This project is released under the GPL-3.0 License License.
  * 
@@ -30,7 +30,7 @@ namespace AS5047P_Types {
         );
     }
 
-    #if defined(ARDUINO_ARCH_SAMD) || defined(CORE_TEENSY)
+    #if defined(AS5047P_STD_STRING_SUPPORT)
 
     std::string ERROR_t::toStdString() {
         

--- a/src/types/AS5047P_Types.h
+++ b/src/types/AS5047P_Types.h
@@ -2,8 +2,8 @@
  * @file AS5047P_Types.h
  * @author Jonas Merkle [JJM] (jonas@jjm.one)
  * @brief This headerfile contains type definitions for the AS5047P Library.
- * @version 2.1.5
- * @date 2021-04-10
+ * @version 2.2.0
+ * @date 2022-11-20
  * 
  * @copyright Copyright (c) 2021 Jonas Merkle. This project is released under the GPL-3.0 License License.
  * 
@@ -14,7 +14,9 @@
 
 #include <inttypes.h>
 
-#if defined(ARDUINO_ARCH_SAMD) || defined(CORE_TEENSY)
+#include "./util/AS5047P_Settings.h"
+
+#if defined(AS5047P_STD_STRING_SUPPORT)
 #include <string>
 #endif
 
@@ -131,7 +133,7 @@ namespace AS5047P_Types {
              */
             bool noError();
 
-            #if defined(ARDUINO_ARCH_SAMD) || defined(CORE_TEENSY)
+            #if defined(AS5047P_STD_STRING_SUPPORT)
 
             /**
              * Converts the error information into an human readable string.

--- a/src/util/AS5047P_Settings.h
+++ b/src/util/AS5047P_Settings.h
@@ -2,8 +2,8 @@
  * @file AS5047P_Settings.h
  * @author Jonas Merkle [JJM] (jonas@jjm.one)
  * @brief This headerfile contains settings information for the AS5047P Library.
- * @version 2.1.5
- * @date 2021-04-10
+ * @version 2.2.0
+ * @date 2022-11-20
  * 
  * @copyright Copyright (c) 2021 Jonas Merkle. This project is released under the GPL-3.0 License License.
  * 
@@ -31,5 +31,15 @@
  * 
  */
 //#define AS5047P_SPI_ARDUINO_INIT_ON_COM_ENAB
+
+/**
+ * @brief Uncomment this to use std::string compatible text output. 
+ * 
+ * This way of string handling is currently not supported by the default Arduino IDE an will result in errors during compilation.
+ * Pleas use an alternative IDE like PlatformIO + VSCode.
+ * 
+ */
+//#define AS5047P_STD_STRING_SUPPORT
+
 
 #endif // AS5047P_Settings_h

--- a/src/util/AS5047P_Util.h
+++ b/src/util/AS5047P_Util.h
@@ -2,8 +2,8 @@
  * @file AS5047P_Util.h
  * @author Jonas Merkle [JJM] (jonas@jjm.one)
  * @brief This headerfile contains util functions for the AS5047P Library.
- * @version 2.1.5
- * @date 2021-04-10
+ * @version 2.2.0
+ * @date 2022-11-20
  * 
  * @copyright Copyright (c) 2021 Jonas Merkle. This project is released under the GPL-3.0 License License.
  * 
@@ -14,7 +14,9 @@
 
 #include <inttypes.h>
 
-#if defined(ARDUINO_ARCH_SAMD) || defined(CORE_TEENSY)
+#include "./util/AS5047P_Settings.h"
+
+#if defined(AS5047P_STD_STRING_SUPPORT)
 #include <string>
 #include <sstream>
 #endif
@@ -51,7 +53,7 @@ namespace AS5047P_Util {
 
     }
 
-    #if defined(ARDUINO_ARCH_SAMD) || defined(CORE_TEENSY)
+    #if defined(AS5047P_STD_STRING_SUPPORT)
 
     /**
      * @brief Convert a value to a string (see std::to_string)

